### DISCUSSION TITLE
NI-VISA Support planned?

### DISCUSSION THREAD

**Discussion by helblingVHB** [ni/grpc-device#728] [Q&A] [answered]
Hello!

Is it planned to support NI-VISA with grpc-device? Would be very helpful.

Thank you!

**Comment by supriyarao92** [comment, marked as answer]
Hello. 

NI-VISA support is on our roadmap currently being targeted for 2nd half of 2023. We are still gathering active feedback from customers who are interested to see this functionality supported. Would you be interested to share more details on your requirements to use Ni-VISA with gRPC device? I am happy to schedule sometime with you in next couple of weeks.

**Reply by helblingVHB** [reply]
Dear Mrs. Kativarapu,

thank you for your reply!
In the meantime, we realized that we can fulfill our requirements by directly accessing the power supply using TCP sockets, circumventing the need for VISA in this case. However, VISA support by grpc-device might still be a nice feature for the future.
We are currently using grpc-device for DAQmx and XNET devices, and it works like a charm! Excellent work 👍

**Reply by supriyarao92** [reply]
Thanks for sharing your feedback. I will pass it on to the team.

**Reply by Blackclaws** [reply]
Is this already implemented somewhere? I need to access NI-VISA compatible devices from Keysight but I can't in 64 bit applications for some reason. I wanted to use gRPC locally to circumvent that issue.

**Reply by kashbaugh-ni** [reply]
Yes, grpc-device supports NI-VISA as of the 2.4 release. (Just updated the readme to reflect the support.) We do not distribute 32-bit builds, but users are free to take the source and build their own 32-bit server to suit their needs.

**Comment by supriyarao92** [comment]
Hi @helblingVHB. Hope you are having a great day. 
I was wondering if you would be willing to join a quick 30 minute call to share your feedback in detail about your project, customer and how you are using DAQmx and XNET currently, how do you plan to use VISA once available? This information will help us improve the ongoing gRPC related development we are doing. Please send me an email if you would be interested to chat with us.
Here is my email address: **supriya.kativarapu@ni.com**

**Comment by Livius90** [comment]
NI-VISA would be an essential component because in a general measurement system, usually, there are some other 3rd party SCPI, Ivi based devices. They could be used in NI gRPC nicely with combination of gRPC supported NI devices.
